### PR TITLE
Install rspec

### PR DESCRIPTION
**Pull request by meganniu**
The following are installed in addition to `rspec`:
+ factory bot rails
+ database cleaner

Descriptions of file changes follow:
+ `Gemfile`
  +  specified gems for `rspec-rails`, `factory_robot_rails`, and `database_cleaner`
+ `test` directory
  + removed `test` dir and replaced by `rspec` dir when `rspec` was installed
+ `spec/spec_helper.rb`
  + auto-generated file when `rspec` was installed
+ `spec/rails_helper.rb`
  + configured file to use `database_cleaner` and `factory_bot_rails` in tests
  + run it  from root using `rspec spec/hello_world_spec.rb`
+ `spec/hello_world_spec.rb`
  + example of `rspec` usage

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -56,13 +56,30 @@ GEM
       minitest (~> 5.1)
       tzinfo (~> 1.1)
       zeitwerk (~> 2.1, >= 2.1.8)
+    addressable (2.7.0)
+      public_suffix (>= 2.0.2, < 5.0)
     bootsnap (1.4.5)
       msgpack (~> 1.0)
     builder (3.2.3)
     byebug (11.0.1)
+    capybara (3.29.0)
```
Comment (ddgwilli) [inline Gemfile.lock]: I don't think we need this capybara gem?

**Comment by ddgwilli** [comment]
Could you please document the changes in each file so that it's easier for everyone to review?

File: Gemfile.lock [hunk 2]
```diff
@@ -56,13 +56,30 @@ GEM
       minitest (~> 5.1)
```
Comment (ddgwilli) [inline Gemfile.lock]: Looks like there are a lot of new packages in here that look like they're coming out of no-where. Is stuff like `addressable` and `public-suffix` needed?

File: Gemfile.lock (hunk 1 above)
Comment (meganniu) [reply Gemfile.lock]: You're right, going to remove it.

File: Gemfile.lock (hunk 2 above)
Comment (meganniu) [reply Gemfile.lock]: There are definitely alot of useless gems in this file. I think they were added when I installed capybara (before I checked with you if we needed capybara). I've removed everything that was added for capybara.

File: spec/hello_world_spec.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+def greeting
+  'Hello world!'
+end
+
+describe 'Greeting' do
+  it 'returns message' do
+    message = greeting()
```
Comment (ddgwilli) [inline spec/hello_world_spec.rb]: You don't need parenthesis to call methods that don't take parameters in ruby.

```suggestion
	    message = greeting
```

**Pull request merged**